apps/src/code-studio/pd/foorm/editor/FoormEditorPreview.jsx

```diff
@@ -0,0 +1,68 @@
+// Preview panel for Foorm editor.
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import Foorm from '../Foorm';
+import {connect} from 'react-redux';
+
+const styles = {
+  previewBox: {
+    border: '1px solid #eee'
+  },
+  errorMessage: {
+    fontWeight: 'bold',
+    padding: '1em'
+  }
+};
+
+class FoormEditorPreview extends Component {
+  static propTypes = {
+    formHasError: PropTypes.bool,
```

Comment:
maybe pull formHasError to the bottom and mark as from redux like you did in other places?

Reply:
good catch